app/(pages)/_contexts/SelectModeContext.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useContext, useState, ReactNode } from 'react';
+
+interface SelectModeContextType {
+  selectMode: boolean;
+  toggleSelectMode: () => void;
+  setSelectMode: (mode: boolean) => void;
+}
+
+const SelectModeContext = createContext<SelectModeContextType | undefined>(
+  undefined
+);
+
+export const useSelectMode = () => {
```

Comment:
define this in the _hooks folder, you can remove this from the context provider file